Create RSD section

mingoland opened the pull request:
# Changes in this PR
- Create a new section for RSD
- Create an entry point on the Design System page
- Create an `Our approach` page
- Create a design pattern page for `Footer`
- Create a design pattern page for `Contact details`
- Create a design pattern page for `Task guidance`
- Create a design pattern page for `Page not found`

## Screenshots of UI changes

### Entry point on the Design System page
<img width="1265" alt="Screenshot 2024-07-09 at 15 16 37" src="https://github.com/DFE-Digital/design/assets/6421298/140621e5-0fe2-4c52-b43f-140228f90a3e">

### Landing page of the RSD section
![screencapture-design-system-rsd-c10b0db1a3a1-herokuapp-design-system-rsd-design-system-2024-07-09-15_17_07](https://github.com/DFE-Digital/design/assets/6421298/6f11d8fc-559c-4c96-bc8d-76cfd924ca89)

nwarms committed:
Add RSD sesign system page

nwarms committed:
Add RSD design pattern page

nwarms committed:
Add the footer with code frame example

nwarms committed:
Update side nav and design system to access the RSD page

nwarms committed:
Update static assets

mingoland committed:
Create index.html

Creating a new contact details page by using the footer page as an example. Content are copied partially only as I take it as a testing only. 

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Create example.html for Contact details

Copying the structure of the example.html from the Footer pattern. Content will be updated later. 

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Update _side-nav.html

Adding Contact Details to the sidenav of RSD

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Update the RSD sidenav

Added trust details and task guidance to the RSD sidenav

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Removed an unnecessary div from footer / index.html

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Removed an unnecessary div from contact-details / index.html

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Create example.html for Trust Details

Signed-off-by: mingoland <mingosoft@gmail.com>

mingoland committed:
Create index.html for Trust details

Signed-off-by: mingoland <mingosoft@gmail.com>

adyhoran1 committed:
Update contact details index.html

Updated the page content based on wiki page: https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_wiki/wikis/Academies-and-Free-Schools-SIP.wiki?wikiVersion=GBwikiMaster&_a=edit&pagePath=/Regional%20Services%20Division%20(RSD)/Cross%252Dcutting%20UCD/Design%20patterns/Contact%20details&pageId=7376

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update design patterns index.html

Updated design patterns page content.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contacts index.html

Removed an </h2> that I'd added in error.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update _side-nav.html

Changed Design Patterns to Design patterns.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update content index.html

Shortened page description.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update content index.html

Shortened page description further.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Create example.html for task guidance

Signed-off-by: mingoland <mingosoft@gmail.com>

adyhoran1 committed:
Update trust details index.html

Updated trust details index to match design pattern.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Create index.html for Task guidance

Signed-off-by: mingoland <mingosoft@gmail.com>

adyhoran1 committed:
Update task guidance index.html

Updated page content based on wiki entry.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Upload images to app - assets

Images for Regional Services Division design system.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Uploaded images to public - assets

Images for Regional Services Division design system.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Update example.html for Contact details

Updated the rendered html 

Signed-off-by: mingoland <mingosoft@gmail.com>

adyhoran1 committed:
Rename Contacts - St George's Primary School.png to contacts-st-georges-primary-school.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Contacts - St George's Primary School.png to contacts-st-georges-primary-school

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Edit contact.png to edit-contact.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Edit contact.png to edit-contact.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename External contacts.png to external-contacts.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename External contacts.png to external-contacts.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Is this the correct trust.png to is-this-the-correct-trust.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Is this the correct trust.png to is-this-the-correct-trust.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename TRN (trust reference number).png to trn-trust-reference-number.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename TRN (trust reference number).png to trn-trust-reference-number.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Update HTML snippet in contact details index.html

Signed-off-by: mingoland <mingosoft@gmail.com>

adyhoran1 committed:
Rename Task guidance, details component expanded.png to task-guidance-details-component-expanded.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Task guidance, details component expanded.png to task-guidance-details-component-expanded.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Task guidance, details component hidden.png to task-guidance-details-component-hidden.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Task guidance, details component hidden.png to task-guidance-details-component-hidden.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Trust summary cards.png to trust-summary-cards.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Trust summary cards.png to trust-summary-cards.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Trust summary list.png to trust-summary-list.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Rename Trust summary list.png to trust-summary-list.png

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Added images to contact details index.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update trust details index.html

Added images to trust details.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update task guidance index.html

Update task guidance content and add images.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update task guidance index.html

Update file paths of images.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Updated file path of images.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update trust details index.html

Added images to trust details page.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update trust details index.html

Updated alt text.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Updated alt class on images.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update task guidance index.html

Updted alt class on images.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Added the second snippet to Contact Details

adyhoran1 committed:
Update footer index.html

Updated footer index with new page content.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Updating footer snippet and screengrab

adyhoran1 committed:
Update footer index.html

Updated page content.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Introducing multiple snippets to the contact details design

mingoland committed:
Added snippet to task guidance

mingoland committed:
Re-factored task guidance

mingoland committed:
Added placeholder for search for an establishment

adyhoran1 committed:
Update design patterns index.html

Changed GDS design system to GOV.UK design system

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Moved code snippets around and added content.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Fixed paragraph formatting.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Simplified wording on contact details page.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

nwarms committed:
Update the code frame index.html to include nunjucks snippets

nwarms committed:
Update footer index.html to include the nunjucks snippet

nwarms committed:
Update RSD index.html main text to be more desciptive
Taken and influenced by the NCS page

adyhoran1 committed:
Update footer index.html

Tweaked the content now that we have nunjucks code.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

mingoland committed:
Added a placeholder page for the page not found pattern

adyhoran1 committed:
Update _side-nav.html

Changed 'Page not found pages' to 'Page not found'

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update contact details index.html

Changed read only per card to read only per row, in intro snippet.

Signed-off-by: adyhoran1 <97228674+adyhoran1@users.noreply.github.com>

adyhoran1 committed:
Update _side-nav.html

Changed Design patterns to Our approach

adyhoran1 committed:
Fixing broken sidebar link.

mingoland committed:
Added the read-only design back to Contact Details

adyhoran1 committed:
Update contact details indexed

Re-jigged snippets.

adyhoran1 committed:
Reviewed homepage content

Fixed typo and updated content.

adyhoran1 committed:
Update index.html

Made it more explicit why users should use the contact details page.

adyhoran1 committed:
Update index.html

Fixed h4 issue.

adyhoran1 committed:
Update page not found content

Updated page not found content so it fits RSD's context.

adyhoran1 committed:
Update index.html

Tweaked the content to improve readability.

nwarms committed:
Update contact details index.html

Stub out nunjucks

Change name of first code frame to avoid clash

nwarms committed:
Update code-frame.html hide nunjucks properly when there is no nunjucks

mingoland committed:
Fixing the our approach broken link

adyhoran1 committed:
Update index.html

Fixed typo.

adyhoran1 committed:
Update index.html

Fixed typo.

mingoland committed:
updating structure of search for establishment

adyhoran1 committed:
Contact details

Added more relatable content to coded examples.

adyhoran1 committed:
Update index.html

Changed content per row so it's easier to understand.

adyhoran1 committed:
Fixed examples

Fixed examples so they're consistent.

adyhoran1 committed:
Update index.html

Removed code snippet from bottom of page.

adyhoran1 committed:
Update index.html

Tweaked content so it makes sense.

adyhoran1 committed:
Update index.html

Fixed heading structure so it makes sense.

adyhoran1 committed:
Update index.html

Updated content.

mingoland committed:
Added values to the edit pages

adyhoran1 committed:
Update index.html

Updated the content so it makes sense to anyone on the internet.

adyhoran1 committed:
Update index.html

Fixed typos.

nwarms committed:
Update _side-nav.html to split the rsd section into 2 lists with headers

nwarms committed:
Create code-snippets.html to store the html and nunjucks code snippets

nwarms committed:
Update contact details index.html to use the code snippets instead of having code set in this file

nwarms committed:
Update page not found index.html nunjucks snippet

nwarms committed:
Updated page not found example.html to have built in header and footer

adyhoran1 committed:
Changed casing of Design Patterns

nwarms committed:
Update task guidance index.html
Add nunjucks snippet

adyhoran1 committed:
Update _side-nav.html

Reduced side bar options.

nwarms committed:
Update code-snippets.html
Add nunjucks for edit contact forms

adyhoran1 committed:
Search for an establishment

Removed search for an establishment from sidebar.

mingoland committed:
Updating dummy content for Nunjucks snippets

nwarms committed:
Update contact details code-snippets.html
Add imports for the govuk input to the nunjuck edit templates

mingoland committed:
Fixed bugs in nunjucks snippet

adyhoran1 committed:
Removed h2

Removed rsd patterns h2

adyhoran1 committed:
Headings

Changed heading classes so they are correct.

adyhoran1 committed:
Headings

Changed heading classes so they are correct.

adyhoran1 committed:
Update index.html

Added extra user needs to contact details design pattern.

nwarms committed:
Update task guidance index.html
Change the nunjucks snippet to use the hint attribute rather than fully custom html

adyhoran1 committed:
Footer nunjucks

Updated the nunjucks content for the footer.

mingoland committed:
Updated task guidance html and footer nunjucks tab

mingoland committed:
Making the Nunjucks missing message generic

mingoland commented:
Hi @andyjones81 , may I know if the PR review is in progress? Please let me know if you need further information from us. Thank you.

adyhoran1 committed:
Footer updates

Updated the footer based on feedback from Steve Milnes and devs in 2 RSD teams.

adyhoran1 committed:
Update index.html

Changed How it works to Links

adyhoran1 committed:
Update index.html

Removed opens in new tab.

mingoland committed:
Making the footer example and HTML code consistent

andyjones81 commented on app/views/design-system/rsd-design-system/index.html:
```diff
@@ -0,0 +1,20 @@
+{% block beforeContent %}{% endblock %}
+
+{% block content %}
+<p>You can use the Regional Services Division (RSD) design system to help you design and build <a href="https://design-histories.education.gov.uk/team/regional-services-division" target="_blank" rel="noopener noreferrer">RSD products</a>.
```
Change this so that it includes the content "(Page on DfE Intranet opens in new tab)" as either visible or govuk-visually-hidden text in a span

adyhoran1 replied:
@andyjones81 are you referring to this link: https://design-histories.education.gov.uk/team/regional-services-division

If so, I believe this should open in the same tab as it's a DfE Design Manual page.

mingoland replied:
I have just removed the `target="_blank"` so it will not open in a new tab. Does it solve the problem?

andyjones81 commented on app/views/design-system/rsd-design-system/index.html:
```diff
@@ -0,0 +1,20 @@
+<p>You can use the Regional Services Division (RSD) design system to help you design and build <a href="https://design-histories.education.gov.uk/team/regional-services-division" target="_blank" rel="noopener noreferrer">RSD products</a>.
+<p>This will help ensure we create a consistent experience for users who use multiple products.</p>
+<h2>GOV.UK and DfE Design System</h2>
+<p>If there's anything that does not exist in the RSD design system, you can use the <a href="https://design-system.service.gov.uk/get-started/" target="_blank" rel="noopener noreferrer">GOV.UK Design System</a> and <a href="/design-system/">DfE Design System</a>.</p>
```
Link opens in new tab needed, or ideally, follow our style guide and don't open it in a new tab.

mingoland replied:
We changed it now and it will open in the same tab.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+        href="https://forms.office.com/Pages/ResponsePage.aspx?id=yXfS-grGoU2187O4s0qC-SZtRygfwTNOqcfRq-MXpv9UOTIyQlNYR0hJT1Q0TUFVSlJGVFhES01LVC4u">Give
+        feedback about a Regions Group system - Microsoft Form</a>.
+    </p>
+    <h3>Suggest a change (opens in a new tab)</h3>
```
Why is there open in new tab here?

adyhoran1 replied:
This is to match the text links for the (opens in new tab) links in the Example and HTML code.

I believe as these headings sit underneath the Links h2, they make sense out of context if a user was moving between headings. Let me know what you think 🙂

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+        transfers</a>. </p>
+    <p>If you don't have SharePoint guidance about how to use your product, you do not need to include this
+      hyperlink.</p>
+    <h3>Give feedback (opens in a new tab)</h3>
```
Why is there open in new tab here?

adyhoran1 replied:
See my reply about text links.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+    <p>For example, Prepare conversions and transfers: support query</p>
+    <p> It has this text underneath the link: "We aim to respond within 5 working days, or one working day for
+      more urgent queries." </p>
+    <h3>How to use this system (opens in a new tab)</h3>
```
Why is there open in new tab here?

adyhoran1 replied:
See my reply about text links.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+    <h3>How to use this system (opens in a new tab)</h3>
+    <p> This links to the section of a product's SharePoint page with guidance on how to use the product. For
+      example, <a
+        href="https://educationgovuk.sharepoint.com/sites/lvewp00299/SitePages/Prepare-Conversions-and-Transfers.aspx#guidance-for-using-prepare-conversions-and-transfers">Guidance
```
In the "Our approach" page you open the intranet pages in a new tab, do the same here so it's consistent, and also use the hidden / visible text to say it's an intranet page being opened. (as it's only available to DfE staff).

mingoland replied:
We updated the text for all "User needs database" links so that it contains the words "(opens in DfE Microsoft List)" now.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+    <h3>Give feedback (opens in a new tab)</h3>
+    <p>This links to a Microsoft Form where users can give feedback about the product. The same Microsoft Form
+      is used across all RSD products: <a
+        href="https://forms.office.com/Pages/ResponsePage.aspx?id=yXfS-grGoU2187O4s0qC-SZtRygfwTNOqcfRq-MXpv9UOTIyQlNYR0hJT1Q0TUFVSlJGVFhES01LVC4u">Give
```
Consider opening in a new tab

mingoland replied:
Thanks Andy. We had a discussion and since we decided not to open in a new tab for the "User needs database" (an internal Microsoft link), so for consistency, we also decided not to open this one in a new tab.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+    <h2>Research on this component</h2>
+    <p>
+      We have not done research on this component, but it is being used to meet these needs from our <a
+        href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user
```
Internal link again

mingoland replied:
We added the text "(opens in DfE Microsoft List)" now.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/index.html:
```diff
@@ -0,0 +1,64 @@
+    consistency may help them. </p>
+<h2>Rolling-out design patterns across RSD</h2>
+<h3>When designing something new or improving an existing design</h3>
+<p>We check the <a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database</a> to see if a user need we have identified is
```
Internal link again

mingoland replied:
In the updated version, this part is now:

`<a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database (opens on DfE intranet)</a>`

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/page-not-found/index.html:
```diff
@@ -0,0 +1,127 @@
+    <p>
+      We have not done research on this pattern, but it is being used in <a
+      href="https://design-histories.education.gov.uk/manage-free-school-projects">Manage free school projects</a> to meet this need from our <a
+        href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user
```
Internal link again

mingoland replied:
In the updated version, this part is now:

`
We have not done research on this pattern, but it is being used in <a href="https://design-histories.education.gov.uk/manage-free-school-projects">Manage free school projects</a> to meet this need from our <a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database (opens in DfE Microsoft List)</a>
`

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+        <h2>How to use this component</h2>
+        <p>If displaying trust details on a standalone page, use the <a href="https://design-system.service.gov.uk/components/summary-list/">summary list component</a>.
+
+        <img src="public/assets/images/trust-summary-list.png" alt class="A summary list with details about a trust.">
```
You've put alt text in the class attribute. Also make sure the alt text is properly describing the image, and consider using the HTML to render it instead of using an image.

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+      
+        <p>If displaying trust details alongside other summary lists on a page, use the <a href="https://design-system.service.gov.uk/components/summary-list/#summary-cards">summary card</a> design.</p>
+
+        <img src="public/assets/images/trust-summary-cards.png" alt class="2 summary cards. One is called Trust details, and another is called Reference numbers.">
```
You've put alt text in the class attribute. Also make sure the alt text is properly describing the image, and consider using the HTML to render it instead of using an image.

mingoland replied:
Thanks for spotting it. This page is not published yet (no links can access it) so we will fix it later.

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+        
+        <p>If this is the case, you can use a text input box and primary action button.</p>
+
+        <img src="public/assets/images/trn-trust-reference-number.png" alt class="A text input for TRN (trust reference number) with a primary action button.">
```
You've put alt text in the class attribute. Also make sure the alt text is properly describing the image, and consider using the HTML to render it instead of using an image.

mingoland replied:
Same as above.

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+        
+        <p>When users search for the trust, provide a page to allow users to confirm the retrieved information matches what they need.</p>
+
+         <img src="public/assets/images/is-this-the-correct-trust.png" alt class="A table with trust details. Underneath the table is a radio button question, Is this the correct trust?.">
```
You've put alt text in the class attribute. Also make sure the alt text is properly describing the image, and consider using the HTML to render it instead of using an image.

mingoland replied:
Same as above.

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+        
+        <p>There could be trust information that needs to be entered manually, rather than retrieved from a search. For example, the name and email address of a contact in the trust.</p>
+        
+        <p>In this case, add extra rows in your summary list or card, and let users enter the information manually. See the <a href="https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_wiki/wikis/Academies-and-Free-Schools-SIP.wiki/7376/Contact-details">contact details design pattern</a>.</p>
```
This is a link to devops which is for DfE staff only. Make sure there is content to say this is for DfE staff and consider a new tab (as it's 'internal only')

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/trust-details/index.html:
```diff
@@ -0,0 +1,149 @@
+        
+        <h2>Research on this pattern</h2>
+        
+        <p>We have not done research on this design pattern, but the pattern is being used to meet these needs from our <a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database</a>:</p>
```
Intranet link

mingoland replied:
Thanks for flagging it. We have just removed the entire section for now but we will definitely pay attention to that problem in other parts of the website.

andyjones81 commented on app/views/design-system/rsd-design-system/index.html:
```diff
@@ -0,0 +1,20 @@
+<p>You can use the Regional Services Division (RSD) design system to help you design and build <a href="https://design-histories.education.gov.uk/team/regional-services-division" target="_blank" rel="noopener noreferrer">RSD products</a>.
+<p>This will help ensure we create a consistent experience for users who use multiple products.</p>
+<h2>GOV.UK and DfE Design System</h2>
+<p>If there's anything that does not exist in the RSD design system, you can use the <a href="https://design-system.service.gov.uk/get-started/" target="_blank" rel="noopener noreferrer">GOV.UK Design System</a> and <a href="/design-system/">DfE Design System</a>.</p>
```
Don't need the trailing / here. Just use "/design-system"

mingoland replied:
Thanks Andy the trailing / is removed.

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/code-frame.html:
```diff
@@ -0,0 +1,46 @@
+  <div class="govuk-tabs__panel" id="nunjucks-code-tabs__panel-{{ id }}">
+    We have not been able to create Nunjucks for this pattern.
+
+    If you have any problems implementing this pattern, ask in the <a href="https://ukgovernmentdfe.slack.com/archives/C04N7UJHRHU" target="_blank">#rsd-ucd-community-public</a> Slack channel. 
```
all target="_blank" links need to say "Open in new tab" either hidden or visually and, must include the rel="noopener noreferrer" attribute

For links which only DfE staff can access (slack, intranet, devops) make sure it says it's for DfE staff only or similar.

mingoland replied:
Added the rel="noopener noreferrer" attribute

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/example-contact-per-card-edit.html:
```diff
@@ -0,0 +1,26 @@
+    <label class="govuk-label govuk-heading-m" for="teamLeadEmailAddress">
+      Email address
+    </label>
+    <input class="govuk-input" id="teamLeadEmailAddress" name="teamLeadEmailAddress" type="text" value="alex.james@education.gov.uk">
```
Don't put potentially real user email addresses  into this code. remove this.

mingoland replied:
Thanks Andy, we have replaced all @education.gov.uk emails with @example.com

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/example-contact-per-card-read-only.html:
```diff
@@ -0,0 +1,63 @@
+          Email address
+        </dt>
+        <dd class="govuk-summary-list__value">
+          alex.james@education.gov.uk
```
again, don't put emails here, use something generic like example.name@education.gov.uk

mingoland replied:
We have replaced all @education.gov.uk emails with @example.com

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/example-contact-per-card-read-only.html:
```diff
@@ -0,0 +1,63 @@
+          Email address
+        </dt>
+        <dd class="govuk-summary-list__value">
+          zara.laney@education.gov.uk
```
See previous comments about email's

mingoland replied:
We have replaced all @education.gov.uk emails with @example.com

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/example-contact-per-row-read-only.html:
```diff
@@ -0,0 +1,76 @@
+        </dt>
+        <dd class="govuk-summary-list__value">
+          Alex James<br/>
+          alex.james@education.gov.uk
```
See previous comments about emails

mingoland replied:
We have replaced all @education.gov.uk emails with @example.com

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/example-contact-per-row-read-only.html:
```diff
@@ -0,0 +1,76 @@
+        </dt>
+        <dd class="govuk-summary-list__value">
+          Zara Laney<br/>
+          zara.laney@education.gov.uk
```
See previous comments about emails

mingoland replied:
We have replaced all @education.gov.uk emails with @example.com

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/contact-details/index.html:
```diff
@@ -0,0 +1,121 @@
+          codeSnippet.htmlCodeContactPerCardEditable, codeSnippet.nunjucksCodeContactPerCardEditable) }}
+        
+        <h2>Research on this pattern</h2>
+        <p>We have not done research on this design pattern, but the pattern is being used in <a href="https://design-histories.education.gov.uk/prepare-conversions-transfers">Prepare conversions and transfers</a> and <a href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete conversions, transfers and changes</a> to meet these user needs from our <a h ...
```
See previous comments about internal links

mingoland replied:
This part is now:

```
<p>We have not done research on this design pattern, but the pattern is being used in <a href="https://design-histories.education.gov.uk/prepare-conversions-transfers">Prepare conversions and transfers</a> and <a href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete conversions, transfers and changes</a> to meet these user needs from our <a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database (opens in DfE Microsoft List)</a>:</p>

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,176 @@
+          an academy (A2B)</a> This site is for external users and the Microsoft links in this pattern
+        won't work for them</li>
+      <li>SharePoint guidance pages, like <a
+          href="https://educationgovuk.sharepoint.com/sites/lvewp00299/SitePages/Prepare-Conversions-and-Transfers.aspx">Prepare
```
See previous comments about internal links

mingoland replied:
Now this part is like that:

```
<li>SharePoint guidance pages, like <a
          href="https://educationgovuk.sharepoint.com/sites/lvewp00299/SitePages/Prepare-Conversions-and-Transfers.aspx">Prepare
          conversions and transfers (opens on DfE intranet)</a></li>

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/search-for-an-establishment/index.html:
```diff
@@ -0,0 +1,238 @@
+        <p>read-only start page with the updated result</p>
+
+        <h2>Research on this pattern</h2>
+        <p>We have not done research on this design pattern, but the pattern is being used in <a href="https://design-histories.education.gov.uk/prepare-conversions-transfers">Prepare conversions and transfers</a> and <a href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete conversions, transfers and changes</a> to meet this need from our <a href="ht ...
```
See comments about internal links

mingoland replied:
Updated to:

```
<p>We have not done research on this design pattern, but the pattern is being used in <a href="https://design-histories.education.gov.uk/prepare-conversions-transfers">Prepare conversions and transfers</a> and <a href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete conversions, transfers and changes</a> to meet this need from our <a href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user needs database (opens in DfE Microsoft List)</a>:</p>

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/task-guidance/index.html:
```diff
@@ -0,0 +1,231 @@
+        <p>The pattern is being used in <a
+            href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete
+            conversions, transfers and changes</a> to meet this need from our <a
+            href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user
```
See comments about internal links

mingoland replied:
Updated to:

```
<p>The pattern is being used in <a
            href="https://design-histories.education.gov.uk/complete-conversions-transfers-and-changes">Complete
            conversions, transfers and changes</a> to meet this need from our <a
            href="https://educationgovuk.sharepoint.com/sites/ServiceDeliveryDirectorate/Lists/RSD%20crossprogramme%20user%20needs/AllItems.aspx?env=WebViewList">user
            needs database (opens in DfE Microsoft List)</a>:</p>

andyjones81 reviewed (changes requested):
Please see comments and make changes as necessary.

mingoland committed:
make the RSD product link staying in the same tab

mingoland committed:
The GOV.UK DS link will now open in the same tab

mingoland committed:
Fixing the differences in the code so that these three links will open in a new tab

mingoland committed:
Making links of user needs database clearer

mingoland committed:
Removing the trailing slash from the link

mingoland committed:
Replaced @education.gov.uk by @example.com

mingoland committed:
Adding a new pattern 'There is a problem with the service'

adyhoran1 committed:
Update example.html

Updated link text in footer.

adyhoran1 committed:
Update index.html

Updated the Email Service Support section, and the h3 link text in the Links section.

adyhoran1 committed:
Update index.html

Fixed formatting issue in guidance.

adyhoran1 committed:
Update index.html

Updated guidance RE (opens in new tab).

adyhoran1 committed:
Update index.html

Updated the point about SharePoint guidance.

adyhoran1 committed:
Update index.html

Fixed opens in new tab typo

adyhoran1 committed:
Update example.html

Updated page not found example.

adyhoran1 committed:
Update example.html

adyhoran1 committed:
Page not found updates

Updated the body copy so it's more specific as an RSD pattern.

adyhoran1 committed:
Content change.

Made the content more concise.

adyhoran1 committed:
Content tweak

Edited the content so it makes sense.

adyhoran1 committed:
Update example.html

Added (opens in a new tab)

adyhoran1 committed:
Update index.html

Added an explanation about 'get help with a et cetera' link.

adyhoran1 committed:
Update index.html

Shortened the bullet point and link text

adyhoran1 committed:
Update index.html

Updated the footer HTML.

adyhoran1 committed:
Update index.html

Updated the HTML link text

adyhoran1 committed:
Update _side-nav.html

Deleted There is a problem page from the side nav.

adyhoran1 committed:
Update index.html

Updated the HTML

adyhoran1 committed:
Footer image

Tested using a different image.

adyhoran1 committed:
Footer image

Added image to 'public' folder.

adyhoran1 committed:
Footer updates

Updated the footer images.

adyhoran1 committed:
Footer image

Further update to footer image.

mingoland committed:
removing the pattern There is a problem

adyhoran1 committed:
Update index.html

Changed the word component to pattern.

adyhoran1 committed:
Internal users

Removed reference of internal and external users - so this pattern could be implemented and tested by other DfE teams.

adyhoran1 committed:
Experienced

Fixed typo in experienced.

mingoland committed:
Removed Trust Details for now

mingoland committed:
adding label for external links on the search for an establishment index page

mingoland committed:
Added noopener noreferrer for a Slack link

andyjones81 commented on app/views/design-system/rsd-design-system/design-patterns/footer/index.html:
```diff
@@ -0,0 +1,174 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+
+    {{ sample.code('/design-system/rsd-design-system/design-patterns/footer/example.html', 'Footer', 'footer',
```
This is loading an iframe which in turn is pulling in the example.html which contains full <html><body> structure which isn't needed here, this is resulting in the example view in the tab applying the width grids etc, within existing grids so looks like it's mobile view. 

Use the full html structure example for the "Open example in a new tab" view, but it's not needed for the preview. I expect this is repeated in all the examples, so revisit them.

nwarms replied:
That makes sense why the components look like that. However this is going to also affect the NCS section since they also use iframes to render their previews

mingoland replied:
Hi @andyjones81 @nwarms , sorry that I'm a bit keen on getting this PR approved and merged, so I applied a temp fix but forcing the iframe to show at 100% width. We can iterate this part later if we want to fix NCS and RSD together. I hope it is okay? 

## Screengrab of the changes made in code:
<img width="1024" alt="Screenshot 2024-10-29 at 13 42 56" src="https://github.com/user-attachments/assets/8bf32212-9c92-4646-b3d1-58c7450da5e2">

## Results:
<img width="1241" alt="Screenshot 2024-10-29 at 13 42 46" src="https://github.com/user-attachments/assets/fae75c26-72e6-4159-afd1-388067b7be33">

andyjones81 reviewed (changes requested):
Se comment (footer index) for how examples have been rendered. This is resulting in multiple grid layouts being applied.

mingoland committed:
Adding a temp fix to iframe
